packages/framework/ai-collab/src/test/explicit-strategy/utils.ts

```diff
@@ -76,3 +78,101 @@ export function initializeOpenAIClient(service: "openai" | "azure"): OpenAI {
 		return client;
 	}
 }
+
+/**
+ * A utility class for snapshot testing.
+ */
+export class MochaSnapshotUnitTester {
+	public constructor(
+		public readonly snapshotDirectory: string,
+		public readonly suiteName: string,
+	) {}
+
+	public expectToMatchSnapshot(
+		mochaContext: Mocha.Context,
+		output: string,
+		options?: { fileNameOverride?: string; metadata?: Record<string, string> },
+	): void {
+		const snapshotBaseFolderDir = path.join(this.snapshotDirectory, "__snapshots__");
+
+		// Create the __snapshots__ directory if it does not exist
+		if (!fs.existsSync(snapshotBaseFolderDir)) {
+			fs.mkdirSync(snapshotBaseFolderDir);
+		}
```

Comment:
I think this might be redundant and we can rely on the call below once we add the suite name. If anything, `mkdirSync` might have extra arguments to ensure it "creates all the way to the bottom".

Reply:
Yep, you're right we can get away with just updating the call below to `fs.mkdirSync(snapshotDir, { recursive: true });`  Updated in commit 6fc87d9de1a64f7fb50d2294ce8a29b8ced9e904